### DISCUSSION TITLE
mdc2 not working on Ubuntu-22.04.x (OpenSSL-3.x.x)

### DISCUSSION THREAD

**Discussion by rumas84** [openssl/openssl#22051] [Q&A] [answered]
```
pc@pc:/tmp/test-mdc$ make
cc  -lcrypto -c -o test.o example.c
example.c: In function ‘main’:
example.c:14:30: warning: implicit declaration of function ‘EVP_mdc2’; did you mean ‘EVP_md5’? [-Wimplicit-function-declaration]
   14 |         EVP_DigestInit_ex(c, EVP_mdc2(), NULL);
      |                              ^~~~~~~~
      |                              EVP_md5
example.c:14:30: warning: passing argument 2 of ‘EVP_DigestInit_ex’ makes pointer from integer without a cast [-Wint-conversion]
   14 |         EVP_DigestInit_ex(c, EVP_mdc2(), NULL);
      |                              ^~~~~~~~~~
      |                              |
      |                              int
In file included from example.c:1:
/usr/include/openssl/evp.h:708:61: note: expected ‘const EVP_MD *’ {aka ‘const struct evp_md_st *’} but argument is of type ‘int’
  708 | __owur int EVP_DigestInit_ex(EVP_MD_CTX *ctx, const EVP_MD *type,
      |                                               ~~~~~~~~~~~~~~^~~~
cc  -o test test.o -lcrypto
/usr/bin/ld: test.o: in function `main':
example.c:(.text+0x56): undefined reference to `EVP_mdc2'
collect2: error: ld returned 1 exit status
make: *** [Makefile:13: test] Error 1
```
so a binary compiled with OpenSSL-3.x.x to work on Ubuntu-22.04.3 
and i have this warning :
`error: /am-***uR/am-***uR: **undefined symbol:** EVP_mdc2, version OPENSSL_3.0.0`

I can't say exactly where the error is on Ubuntu-22.04.x with mcd2,  
But it is certain that mcd2 does not work on ubuntu 22.04.
(... ???)

**Comment by tom-cosgrove-arm** [comment]
Which exact version of OpenSSL are you using? Is it the one that originally came with Ubuntu-22.04.3? Has it been upgraded? Have you installed a more recent version? And how was it configured?

**Comment by nhorman** [comment, marked as answer]
If you're using the Ubuntu provided openssl libs, they explicitly build the library with the no-mdc2 options, so it makes sense you don't find that symbol.  The full config they specify is:
No-idea no-mdc2 no-rc5 no-zlib no-ssl3 enable-unit-test no-ssl3-method enable-rfc3779 enable-cms no-capieng

You can likely ask them to enable mdc2

**Comment by rumas84** [comment]
> Which exact version of OpenSSL are you using?

OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)


> Is it the one that originally came with Ubuntu-22.04.3?

yes,



> Has it been upgraded?

No.


> You can likely ask them to enable mdc2

I will find a way around it without asking them .

Thanks,
